OutOfMemoryError while collecting test output even when disabled

**Description**

I have a build where `showStandardStreams` has the default value of `false`. I get this error:

```
Caused by: java.lang.OutOfMemoryError: Java heap space
	at com.adarshr.gradle.testlogger.logger.OutputCollector.removeTestOutput(OutputCollector.groovy:26)
	at com.adarshr.gradle.testlogger.logger.SequentialTestLogger.afterTest(SequentialTestLogger.groovy:46)
	at com.adarshr.gradle.testlogger.logger.TestLoggerAdapter.afterTest(TestLoggerAdapter.groovy:54)
	at com.adarshr.gradle.testlogger.logger.TestLoggerWrapper.afterTest(TestLoggerWrapper.groovy:12)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:231)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:150)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)
	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:325)
	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:235)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:141)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy88.afterTest(Unknown Source)
	at org.gradle.api.internal.tasks.testing.results.TestListenerAdapter.completed(TestListenerAdapter.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:231)
```

This appears to be caused by the `toString` on the `StringBuffer` consuming enough memory to push past the heap limits. Since the build has the collecting of standard streams disabled this is an unexpected failure. I think a simple solution would be to extend the checking of `showStandardStreams` to the `OutputCollector.collect` and simply leave the StringBuilder empty.

I can work around this by increasing the available heap but I'd prefer the test logger to have lower overhead when not asking for the extra features.

**Versions**

- Test logger version: 2.1.1 (although latest code appears to have same issue) 
- Gradle version: 5.6.4
- Java version: 11.0.8

**Type of test being run**

Whether it is JUnit, Spock, TestNg, or anything else. Also, try to include a sample test here.

**Test logger configuration**

Everything is default so there is no explicit configuration in by gradle build file.

**Additional information**

Add any other information about the problem here.

--- a/src/main/groovy/com/adarshr/gradle/testlogger/logger/TestLoggerAdapter.groovy
+++ b/src/main/groovy/com/adarshr/gradle/testlogger/logger/TestLoggerAdapter.groovy
@@ -59,7 +59,9 @@ class TestLoggerAdapter implements TestLogger {
 
     @Override
     void onOutput(TestDescriptor descriptor, TestOutputEvent outputEvent) {
-        outputCollector.collect(wrap(descriptor), outputEvent.message)
+        if (testLoggerExtension.showStandardStreams) {
+            outputCollector.collect(wrap(descriptor), outputEvent.message)
+        }
     }
 
     private TestDescriptorWrapper wrap(TestDescriptor descriptor) {

--- a/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/ParallelExecutionSpec.groovy
+++ b/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/ParallelExecutionSpec.groovy
@@ -17,6 +17,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                     }
                     testlogger { 
                         theme 'plain-parallel'
+                        slowThreshold 5000
                     }
                  ''',
                 'clean test'
@@ -61,6 +62,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                     testlogger { 
                         theme 'plain-parallel'
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                  ''',
                 'clean test'
@@ -178,6 +180,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                         showSkipped false
                         showFailed false
                         showExceptions false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *SecondSpec'
@@ -212,6 +215,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                         showSkipped false
                         showExceptions false
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -258,11 +262,12 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                         maxParallelForks = gradle.startParameter.maxWorkerCount
                     }
                     testlogger {
-                        theme 'standard-parallel\'
+                        theme 'standard-parallel'
                         showPassed false
                         showSkipped true
                         showFailed false
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *ThirdSpec'
@@ -304,6 +309,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                         theme 'standard-parallel'
                         showStandardStreams true
                         showPassedStandardStreams false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -365,6 +371,7 @@ class ParallelExecutionSpec extends AbstractFunctionalSpec {
                         showStandardStreams true
                         showPassedStandardStreams false
                         showFailedStandardStreams false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'

--- a/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/TestLoggerPluginSpec.groovy
+++ b/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/TestLoggerPluginSpec.groovy
@@ -11,6 +11,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-spock-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *FirstSpec --tests *SecondSpec'
             )
         then:
@@ -86,6 +87,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit4-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *First*'
             )
         then:
@@ -111,6 +113,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit5-vintage-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *First*'
             )
         then:
@@ -136,6 +139,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit5-jupiter-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *First*'
             )
         then:
@@ -161,6 +165,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit5-jupiter-nested-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test'
             )
         then:
@@ -189,6 +194,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit5-jupiter-deep-nested-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test'
             )
         then:
@@ -211,7 +217,12 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-junit5-jupiter-deep-nested-tests',
-                'testlogger { showSimpleNames true }',
+                '''
+                    testlogger { 
+                        showSimpleNames true
+                        slowThreshold 5000
+                    }
+                ''',
                 'clean test'
             )
         then:
@@ -234,6 +245,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-testng-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *First*'
             )
         then:
@@ -262,7 +274,8 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         theme 'plain' 
-                        showSimpleNames true 
+                        showSimpleNames true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test'
@@ -282,6 +295,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'sample-spock-tests',
+                'testlogger { slowThreshold 5000 }',
                 'clean test --tests *SecondSpec*pass'
             )
         then:
@@ -303,6 +317,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         theme 'plain' 
+                        slowThreshold 5000
                     }
                     task anotherTask(type: Test) {
                         useJUnitPlatform()
@@ -349,6 +364,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         showSummary false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test'
@@ -369,6 +385,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test'
@@ -408,6 +425,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *SecondSpec'
@@ -431,6 +449,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                 '''
                     testlogger { 
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -469,6 +488,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                     testlogger { 
                         showPassed false
                         showExceptions false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *SecondSpec'
@@ -507,6 +527,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                         showPassed false
                         showSkipped false
                         showExceptions false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *SecondSpec'
@@ -544,6 +565,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                         showSkipped false
                         showFailed false
                         showExceptions false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *SecondSpec'
@@ -573,6 +595,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                         showSkipped false
                         showExceptions false
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -622,6 +645,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                         showSkipped true
                         showFailed false
                         showStandardStreams true
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec --tests *ThirdSpec'
@@ -661,6 +685,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                     testlogger {
                         showStandardStreams true
                         showPassedStandardStreams false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -720,6 +745,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                         showStandardStreams true
                         showPassedStandardStreams false
                         showFailedStandardStreams false
+                        slowThreshold 5000
                     }
                 ''',
                 'clean test --tests *FirstSpec'
@@ -770,6 +796,7 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                     testlogger {
                         theme 'plain'
                         logLevel 'quiet'
+                        slowThreshold 5000
                     }
                     markerLevel = 'quiet'
                 ''',
@@ -796,12 +823,14 @@ class TestLoggerPluginSpec extends AbstractFunctionalSpec {
                     useJUnitPlatform()
                     testlogger {
                         theme 'plain'
+                        slowThreshold 5000
                     }
                 }
                 task anotherTask(type: Test) {
                     useJUnitPlatform()
                     testlogger {
                         theme 'plain-parallel'
+                        slowThreshold 5000
                     }
                 }
             '''

--- a/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/ThemeSwitchingSpec.groovy
+++ b/src/test-functional/groovy/com/adarshr/gradle/testlogger/functional/ThemeSwitchingSpec.groovy
@@ -11,7 +11,12 @@ class ThemeSwitchingSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'single-spock-test',
-                "testlogger { theme 'plain' }",
+                '''
+                    testlogger { 
+                        theme 'plain'
+                        slowThreshold 5000 
+                    }
+                ''',
                 'clean test'
             )
         then:
@@ -31,7 +36,12 @@ class ThemeSwitchingSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'single-spock-test',
-                "testlogger { theme 'standard' }",
+                '''
+                    testlogger { 
+                        theme 'standard'
+                        slowThreshold 5000 
+                    }
+                ''',
                 'clean test --console plain'
             )
         then:
@@ -50,7 +60,12 @@ class ThemeSwitchingSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'single-spock-test',
-                "testlogger { theme 'standard' }",
+                '''
+                    testlogger { 
+                        theme 'standard'
+                        slowThreshold 5000 
+                    }
+                ''',
                 'clean test'
             )
         then:
@@ -69,7 +84,12 @@ class ThemeSwitchingSpec extends AbstractFunctionalSpec {
         when:
             def result = run(
                 'single-spock-test',
-                "testlogger { theme 'mocha' }",
+                '''
+                    testlogger { 
+                        theme 'mocha'
+                        slowThreshold 5000 
+                    }
+                ''',
                 'clean test'
             )
         then:
@@ -90,7 +110,8 @@ class ThemeSwitchingSpec extends AbstractFunctionalSpec {
                 'single-spock-test',
                 '''
                     testlogger { 
-                        theme 'mocha' 
+                        theme 'mocha'
+                        slowThreshold 5000
                     }
                     test.doLast { 
                         System.clearProperty('testlogger.theme') 

